comp/core/status/component.go

```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package status displays information about the agent.
+package status
+
+import (
+	"io"
+
+	"go.uber.org/fx"
+)
+
+// team: agent-shared-components
+
+// CollectorSection stores the collector section name
+const CollectorSection string = "collector"
+
+// Component interface to access the agent status.
+type Component interface {
+	// Returns all the agent status information for the format type
+	GetStatus(format string, verbose bool) ([]byte, error)
+	// Returns only the agent status for the especify section and format type
+	GetStatusBySection(section string, format string, verbose bool) ([]byte, error)
+}
+
+// Provider interface
+type Provider interface {
+	// Name is used to sort the status providers alphabetically.
+	Name() string
+	// Section is used to group the status providers.
+	Section() string
+	JSON(stats map[string]interface{})
+	Text(buffer io.Writer) error
+	HTML(buffer io.Writer) error
+}
+
+// HeaderProvider interface
+type HeaderProvider interface {
+	// Index is used to choose the order in which the header information is displayed.
+	Index() int
```

Comment:
I still haven't implemented the index using an enum: 

```go 
type HeaderIndex int

const (
  HostnameIndex HeaderIndex = iota
  HostInfoIndex 
  ClocksIndex
  MetadataIndex
)

type HeaderProvider interface {
  Index() HeaderIndex
  ...
}
```

That would allow us to more easily mange the order of the header section. I want to leave this exercise for the future once we start migrating status information into the status component